packages/editor-ui/src/components/RunData.vue

```diff
@@ -533,6 +533,19 @@ export default mixins(
 				this.editMode.value = '';
 			},
 			onClickSaveEdit() {
+				try {
+					JSON.parse(this.editMode.value);
+				} catch (error) {
+					const title = this.$locale.baseText('runData.invalidPinnedData');
+
+					const toRemove = new RegExp(/JSON\.parse:|of the JSON data/, 'g');
+					const message = error.message.replace(toRemove, '').trim();
+					error.message = message.charAt(0).toUpperCase() + message.slice(1);
+
+					this.$showError(error, title);
+					return;
+				}
```

Comment:
Should we also check for a specific structure? Is the value supposed to be an Array of objects?

Reply:
We will be using `pinData` mirroring how `runData` works. I would wait to debug the DS once we connect and can test out how pin data flows.

Reply:
Got it! I'll merge this meanwhile.